File: code_translation/exp1_code_translation_security/translated_solutions_o3/Question4/Java/1038.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class StudentRepository {

    /*  Connection information is normally provided by your environment
        (properties file, environment variables, connection-pool, etc.).
        Placeholders are used here for clarity; replace them with real values. */
    private static final String URL      = "jdbc:mysql://localhost:3306/your_database";
    private static final String USER     = "your_user";
    private static final String PASSWORD = "your_password";

    /**
     * Inserts a student record into the STUDENTS table.
     *
     * @param name the student's name
     * @param age  the student's age
     */
    public static void insertion(String name, int age) {
        String sql = "INSERT INTO STUDENTS (NAME, AGE) VALUES (?, ?)";

        /*  try-with-resources automatically closes the Connection and
            PreparedStatement even if an exception is thrown.            */
        try (Connection conn = DriverManager.getConnection(URL, USER, PASSWORD);
             PreparedStatement ps = conn.prepareStatement(sql)) {

            ps.setString(1, name);
            ps.setInt(2, age);
            ps.executeUpdate();

        } catch (SQLException e) {
            // Re-throw, log, or handle the exception as your project requires.
            e.printStackTrace();
        }
    }
}

File: code_translation/exp1_code_translation_security/translated_solutions_o3/Question4/Rust/1038.rs
use mysql::{params, prelude::Queryable, Pool};
use std::error::Error;

/// Inserts a student row into the STUDENTS table.
///
/// `name` – Student’s name  
/// `age`  – Student’s age
pub fn insertion(name: &str, age: u32) -> Result<(), Box<dyn Error>> {
    // The connection URL will be supplied for you in Cargo.toml / env vars.
    // Example format: mysql://user:password@host:port/database
    let url = std::env::var("DATABASE_URL")?;
    let pool = Pool::new(url)?;
    let mut conn = pool.get_conn()?;

    conn.exec_drop(
        "INSERT INTO STUDENTS (NAME, AGE) VALUES (:name, :age)",
        params! {
            "name" => name,
            "age"  => age,
        },
    )?;

    Ok(())
}

fn main() -> Result<(), Box<dyn Error>> {
    insertion("Alice", 21)?;
    Ok(())
}